src/Middleware/DetectLocaleMiddleware.php

```diff
@@ -84,8 +84,8 @@ public function getLocale(HTTPRequest $request)
             $locale = $this->getPersistLocale($request);
         }
 
-        // Check for locale in browser headers
-        if (empty($locale)) {
+        // Home page only: check for locale in browser headers
+        if (empty($locale) && $request->getURL() === '') {
```

Comment:
Commenting on wrong line, but still in this file;

Shall we move $this->setPersistLocale($request, $locale); to outside of the loop in process? That way we capture the current locale more frequently, in case the next request doesn't have a safe way to guess the locale. (e.g. on immediate redirect).